pkg/cluster/workloadidentityresources.go

```diff
@@ -192,3 +195,71 @@ func (m *manager) getPlatformWorkloadIdentityFederatedCredName(sa string, identi
 
 	return platformworkloadidentity.GetPlatformWorkloadIdentityFederatedCredName(clusterResourceId, identityResourceId, sa), nil
 }
+
+func (m *manager) ensurePlatformWorkloadIdentityRBAC(ctx context.Context) error {
+	resourceGroupID := m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+	resourceGroup := stringutils.LastTokenByte(resourceGroupID, '/')
+
+	var toDelete []mgmtauthorization.RoleAssignment
+	var toAdd []*arm.Resource
+
+	m.log.Infof("retrieving existing role assignments")
+	allExistingRoleAssignments, err := m.roleAssignments.ListForResourceGroup(ctx, resourceGroup, "atScope()")
+	if err != nil {
+		return err
+	}
+	var roleAssignmentsForManagedResourceGroup []mgmtauthorization.RoleAssignment
```

Comment:
Can't we use a map instead here, to save from the extra for loop?

Reply:
Good idea, I'll take a look.